docs/modules/qbi-helper/interest-template.md

```diff
@@ -0,0 +1,141 @@
+```
+
+### Add
+
+When adding queries, you can either provide the queries as parameters or an enumerable set:
+
+```csharp
+// Parameters
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Enumerable
+var queryList = new List() { query1, query2 };
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(queryList);
+```
+
+### Replace
+
+By doing this, you replace all the existing queries for an authoritative component with the new queries you pass in. In a similar way to adding queries, you must provide the queries as parameters or an enumerable set.
+
+```csharp
+// Parameters
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(query1);
+
+// Enumerable
+var queryList = new List() { query1 };
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(queryList);
+```
+
+At least one query must be provided, otherwise the existing queries for a given authoritative component will not be removed.
+
+### Clear
+
+You can either clear queries for a given authoritative component or clear _all_ queries in the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+```csharp
+// Removes all queries added for the Position component
+basicInterestTemplate.ClearQueries<Position.Component>();
+
+// Removes all queries
+basicInterestTemplate.ClearAllQueries();
+```
+
+After adding, removing or modifying a set of queries, there are two ways to get `Interest` out of the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+### Snapshot
+
+To return a [`Snapshot`]({{urlRoot}}/api/core/snapshot) of the `Interest` component that can be used when defining entity templates, call `ToSnapshot()` on the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template).
+
+```csharp
+// Create an interest template
+var interestTemplate = InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Add the Interest component to an entity template
+var entityTemplate = new EntityTemplate();
+...
+entityTemplate.AddComponent(interestTemplate.ToSnapshot(), WorkerUtils.UnityGameLogic);
+```
+
+### ComponentInterest
+
+To update `Interest` at execution time, you need to modify the underlying Dictionary within the `Interest` component.
+
+The [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) class provides the `AsComponentInterest` method to return this Dictionary, so that it can be used to update the `Interest` component.
+
+For example, to update an entity's interest with a completely new set of queries in the MonoBehaviour workflow:
+
+```csharp
+// Create the new interest template
+var newInterestTemplate = InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Require the InterestWriter and update the ComponentInterest field
+InterestWriter.Data.ComponentInterest = newInterestTemplate.AsComponentInterest();
```

Comment:
Don't use this as a demonstration. Its purely reference type semantics that cause the underlying data to change and the component isn't marked as dirty so it may never be replicated.

Instead use the `SendUpdate` method and create an Update object